### PR TITLE
PROVCAU-104 AJAX Summary Displays - Cleanup.

### PR DESCRIPTION
**Pull request by bri-gaia**

### CODE REVIEW

File: app/lib/BaseEditorController.php [hunk 1]
```diff
@@ -699,7 +699,23 @@ public function Summary($pa_options=null) {
 		list($vn_subject_id, $t_subject) = $this->_initView($pa_options);
 		$vb_ajax_load_displays = $t_subject->getAppConfig()->get('enable_ajax_summary_displays');
 		session_write_close();
-		$this->SummaryData($pa_options, $vb_ajax_load_displays);
+		$this->SummaryInfo($pa_options, !$vb_ajax_load_displays || $this->request->isAjax());
+
+		if((defined('__CA_ENABLE_DEBUG_OUTPUT__') && __CA_ENABLE_DEBUG_OUTPUT__) || (bool)$this->request->config->get('display_template_debugger') || ($this->request->user->getPreference('show_template_debugger') !== 'hide')) {
```
Comment (bri-gaia) [inline app/lib/BaseEditorController.php:708]: I think these make sense to place at the page load level, no need to replicate on every ajax request.

File: app/lib/BaseEditorController.php [hunk 2]
```diff
@@ -737,19 +751,24 @@ public function SummaryDisplay($pa_options=null, $pb_ajax_load_displays=FALSE) {
 	 * @param bool $pb_ajax_load_displays Load display using AJAX?
 	 * @return bool
 	 */
-	public function SummaryData($pa_options=null, $pb_ajax_load_displays=FALSE) {
+	public function SummaryData($pa_options=null) {
```
Comment (bri-gaia) [inline app/lib/BaseEditorController.php:760]: Moved reused code into a separate function to make it easier to see which parts are specific to the 'section' we are collecting info for.

File: app/lib/BaseEditorController.php [hunk 3]
```diff
@@ -699,7 +699,23 @@ public function Summary($pa_options=null) {
 		list($vn_subject_id, $t_subject) = $this->_initView($pa_options);
 		$vb_ajax_load_displays = $t_subject->getAppConfig()->get('enable_ajax_summary_displays');
 		session_write_close();
```
Comment (bri-gaia) [inline app/lib/BaseEditorController.php:705]: do we need a session write close here and in summaryInfo() ?

File: app/lib/BaseEditorController.php [hunk 4]
```diff
@@ -737,19 +751,29 @@ public function SummaryDisplay($pa_options=null, $pb_ajax_load_displays=FALSE) {
...
+	}
+	# -------------------------------------------------------
+	/**
+	 * Generates summary of record data based upon a bundle display for screen (HTML)
+	 *
+	 * @param array $pa_options Array of options passed through to _initView
+	 * @param bool $pb_ajax_load_displays Load display using AJAX?
+	 * @return bool
+	 */
+	public function SummaryInfo($pa_options=null, $pb_load_placement_info=TRUE) {
```
Comment (bri-gaia) [inline app/lib/BaseEditorController.php:790]: Use a named variable here to make it clear what it controls, move Ajax-y logic to where this function is called.

File: app/lib/BaseEditorController.php [hunk 5]
```diff
@@ -761,22 +785,10 @@ public function SummaryData($pa_options=null, $pb_ajax_load_displays=FALSE) {
...
-			'url_path' => $this->getRequest()->getFullUrlPath()
-		);
-
-		Session::setVar('save_and_return_locations', caPushToStack($va_save, $va_save_and_return, __CA_SAVE_AND_RETURN_STACK_SIZE__));
-
 		$this->view->setVar('bundle_displays', $va_displays);
 		$this->view->setVar('t_display', $t_display);
 
-		if (!$pb_ajax_load_displays || $this->request->isAjax()) {
+		if ($pb_load_placement_info) {
```
Comment (bri-gaia) [inline app/lib/BaseEditorController.php:810]: If we invert the logic here and return on false, the code is a lot easier to compare to what is upstream (As it no longer needs to be indented, so git can match them up). 
Bit weird though.

File: app/lib/BaseEditorController.php [hunk 6]
```diff
@@ -761,22 +785,10 @@ public function SummaryData($pa_options=null, $pb_ajax_load_displays=FALSE) {
...
 			if ($this->request->isAjax()) {
```
Comment (bri-gaia) [inline app/lib/BaseEditorController.php:811]: Is this the logical place to do this?

File: app/lib/BaseEditorController.php (hunk 3 above)
Comment (kehh) [reply app/lib/BaseEditorController.php:705]: happy for this to go

File: app/lib/BaseEditorController.php (hunk 1 above)
Comment (bri-gaia) [reply app/lib/BaseEditorController.php:708]: Add Access check before rendering this.

**Comment by kehh** [comment]
ah booo conflix

File: app/lib/BaseEditorController.php [hunk 7]
```diff
@@ -721,15 +741,37 @@ public function Summary($pa_options=null) {
...
+	 * Generates summary of record data based upon a bundle display for screen (HTML)
+	 *
+	 * @param array $pa_options Array of options passed through to _initView
+	 *
+	 * @throws ApplicationException
+	 */
+	public function SummaryData($pa_options=null) {
+		$this->SummaryInfo($pa_options, TRUE);
+
+		$ajax_placement_id = $_GET['va_placement_id'];
```
Comment (kehh) [inline app/lib/BaseEditorController.php]: can this be replaced with $this->getRequest()->getParam('va_placement_id') and then use the variable in the setVar below? I know it's from the pre-existing code but use of $_GET should be avoided.

Comment (kehh) [reply app/lib/BaseEditorController.php]: also to follow the CA coding style it should be vn_placement_id or simply placement_id (newer code removes the crazy data typing prefix)

**Review by kehh — CHANGES_REQUESTED** [review]
Tiny tweaks but looking great.

**Pull request merged**